### PR TITLE
Some risc asm improvements

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: riscv/src/compiler.rs [hunk 1]
```diff
@@ -756,40 +766,48 @@ fn process_instruction(instr: &str, args: &[Argument]) -> Vec<String> {
...
-            vec![
-                format!("tmp1 <=X= and({r2}, 0x1f);"),
-                format!("{rd} <=X= shr({r1}, tmp1);"),
-            ]
+            if rd.is_zero() {
+                vec![]
+            } else {
+                vec![
+                    format!("tmp1 <=X= and({r2}, 0x1f);"),
+                    format!("{rd} <=X= shr({r1}, tmp1);"),
```
Comment (leonardoalt) [inline riscv/src/compiler.rs:779]: why not use the new function here too?

File: riscv/src/compiler.rs [hunk 2]
```diff
@@ -756,40 +766,48 @@ fn process_instruction(instr: &str, args: &[Argument]) -> Vec<String> {
...
-            vec![
-                format!("tmp1 <=X= to_signed({rs});"),
-                format!("{rd} <=Y= is_positive({imm} - tmp1);"),
-            ]
+            if rd.is_zero() {
+                vec![]
+            } else {
+                vec![
+                    format!("tmp1 <=X= to_signed({rs});"),
+                    format!("{rd} <=Y= is_positive({imm} - tmp1);"),
```
Comment (leonardoalt) [inline riscv/src/compiler.rs:800]: and here too

**Comment by leonardoalt** [comment]
What happens if it is a write to register 0 though? Is the instruction just ignored?

File: riscv/src/compiler.rs (hunk 1 above)
Comment (chriseth) [reply riscv/src/compiler.rs:779]: Because it only takes a single argument.

**Comment by chriseth** [comment]
Yes, the riscv standard says that the write is ignored.

**Comment by chriseth** [comment]
Ah, this is actually missing the implementation for the memory load instructions.

**Comment by chriseth** [comment]
Updated.

**Review by leonardoalt — APPROVED** [review]

**Comment by leonardoalt** [comment]
Rebased it.

**Pull request merged**